### PR TITLE
Deploys C++ from artifact instead of nightly

### PR DESCRIPTION
**Pull request by StephenNneji**

### CODE REVIEW

File: .github/workflows/runTests.yml [hunk 1]
```diff
@@ -46,6 +44,19 @@ jobs:
...
+      with:
+        command: cppDeploy  
+    - name: Upload cppDelopy
+      if: runner.os == 'Linux'
+      uses: actions/upload-artifact@v4
+      with:
+        name: cppDeploy
+        path: compile/fullCompile/cppDeploy/ 
+    - name: Remove build archive
+      run: rm ../${{ runner.os }}.zip
```
Comment (DrPaulSharp) [inline .github/workflows/runTests.yml:59]: Ahh, is that accounted for here?

File: .github/workflows/runTests.yml [hunk 2]
```diff
@@ -33,9 +33,7 @@ jobs:
       run:  tar --exclude="**/-lang:c++.zip" --exclude=".git*/" --exclude="htmlcov/" -acvf ../${{ runner.os }}.zip *
     - name: Create build archive (Linux)
       if: runner.os == 'Linux'
-      run: |
-        rm -rf ../${{ runner.os }}.zip
-        zip -r ../${{ runner.os }}.zip * -x "**/-lang:c++.zip" ".git*/" "htmlcov/"
+      run: zip -r ../${{ runner.os }}.zip * -x "**/-lang:c++.zip" ".git*/" "htmlcov/*"
```
Comment (DrPaulSharp) [inline .github/workflows/runTests.yml:36]: Does this resolve the merge rather than replace issue?

**Review by DrPaulSharp — COMMENTED** [review]
Probably worth making sure it works with the typos fixed.

File: .github/workflows/runTests.yml (hunk 1 above)
Comment (StephenNneji) [reply .github/workflows/runTests.yml:59]: Yes it is

File: .github/workflows/runTests.yml (hunk 2 above)
Comment (StephenNneji) [reply .github/workflows/runTests.yml:36]: No, the "remove build archive" step does

File: .github/workflows/runTests.yml [hunk 3]
```diff
@@ -44,12 +44,12 @@ jobs:
         overwrite: true
         file: '../${{ runner.os }}.zip'
         asset_name: '${{ runner.os }}.zip'
-    - name: Create cppDelopy
+    - name: Create cppDelpoy
```
Comment (DrPaulSharp) [inline .github/workflows/runTests.yml:47]: The scourge of the off-by-one error.

File: .github/workflows/runTests.yml [hunk 4]
```diff
@@ -44,12 +44,12 @@ jobs:
...
       if: runner.os == 'Linux'
       uses: matlab-actions/run-command@v2
       with:
         command: cppDeploy  
-    - name: Upload cppDelopy
+    - name: Upload cppDelpoy
```
Comment (DrPaulSharp) [inline .github/workflows/runTests.yml:52]: and here

**Review by DrPaulSharp — APPROVED** [review]

**Pull request merged**